### PR TITLE
Force calculation improvement

### PR DESCRIPTION
**Pull request by jonmue42**
Improves the performance of the force calculation by only calculating off site blocks instead of the complete overlap and hamiltonian matrices for the finite differences.

### CODE REVIEW

**Comment by mcsloy** [comment]
Just noticed that the force function is returning an extra dimension for single systems. For a single system I would expect the force matrix to be $n_{atoms}\times 3$ but what I am getting is $1\times n_{atoms}\times 3$ instead. This seems to be coming from the`RepulsiveSplineFeed.gradient` function rather than `_finite_diff_overlap_h0_blocks`.

**Comment by mcsloy** [comment]
I have created a branch with some suggested improvements [here](https://github.com/mcsloy/tbmalt_new/tree/forces_improvement_suggestions). While most suggested changes are superficial there are some minor bugs fixed; such as the use of `torch.tensor` instead of `torch.Tensor`, and device configuration.

**Comment by mcsloy** [comment]
Let me know if you are ready for me to accept this pull request.

**Comment by jonmue42** [comment]
> Let me know if you are ready for me to accept this pull request.

Should now be ready to be accepted.
